src/migrations/migrations-truffle-5/4_deploy_Pool.js

```diff
@@ -33,18 +34,24 @@ async function migrate({
     const contractArtFilePaths = [
       "@gnosis.pm/dx-contracts/build/contracts/DutchExchangeProxy.json",
       "@gnosis.pm/util-contracts/build/contracts/EtherToken.json",
-      "@gnosis.pm/gno-token/build/contracts/TokenGNO.json",
+      process.env.TOKEN_B_ADDRESS ? "@gnosis.pm/util-contracts/build/contracts/HumanFriendlyToken.json" : "@gnosis.pm/gno-token/build/contracts/TokenGNO.json",
     ]
 
     // we need to setProvider to get correct networks from artifacts
     const contractsMapped = contractArtFilePaths.map(path => TC(require(path)))
     // Set deployer provider to each contract
     contractsMapped.forEach(tcArt => tcArt.setProvider(deployer.provider));
-    ([dxProxy, etherToken, tokenGNO] = await Promise.all(contractsMapped.map(contract => contract.deployed())))
+    ([dxProxy, etherToken, tokenB] = await Promise.all(contractsMapped.map(contract => process.env.TOKEN_B_ADDRESS ? contract.at(process.env.TOKEN_B_ADDRESS) : contract.deployed())))
```

Comment:
Assuming `TOKEN_B_ADDRESS` is set, aren't you using `contract.at(process.env.TOKEN_B_ADDRESS)` for all contracts (DutchX, Ether & TokenB) now, whereas you only want it for the last?

Reply:
yeah Anxo caught this too where I was having issues - thanks